### PR TITLE
Propose new sample format.

### PR DESCRIPTION
**Pull request by kurtisvg**
Updated InspectString and InspectFile snippets to the newly proposed format.

### CODE REVIEW

File: dlp/src/main/java/dlp/snippets/InspectFile.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        System.out.println("\tQuote: " + f.getQuote());
+        System.out.println("\tInfo type: " + f.getInfoType().getName());
+        System.out.println("\tLikelihood: " + f.getLikelihood());
+      }
+    } catch (Exception e) {
+      System.out.println("Error during inspectFile: \n" + e.toString());
+    }
+  }
+}
+// [START dlp_inspect_file]
```
Comment (nnegrey) [inline dlp/src/main/java/dlp/snippets/InspectFile.java]: END

File: dlp/src/main/java/dlp.snippets/InspectFile.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        infoTypes.add(InfoType.newBuilder().setName(typeName).build());
+      }
+
+      // Set the inspect configuration for request
+      InspectConfig config = InspectConfig.newBuilder()
+          .addAllInfoTypes(infoTypes)
+          .setIncludeQuote(true)
+          .build();
+
+      // Construct request
```
Comment (lesv) [inline dlp/src/main/java/dlp.snippets/InspectFile.java]: comment seems unhelpful.  The rest are.  Could we say something else or additional?

File: dlp/src/main/java/dlp.snippets/InspectFile.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import com.google.privacy.dlp.v2.ProjectName;
+import com.google.protobuf.ByteString;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class InspectFile {
+
+  // Inspects the specified file.
+  public static void inspectFile(String projectId, String filePath, String fileType) {
```
Comment (lesv) [inline dlp/src/main/java/dlp.snippets/InspectFile.java]: Is this code threadsafe?  (I would suspect it is because the client is created here).  How do I get greater performance?  Should I cache the client(s)?

File: dlp/src/main/java/dlp.snippets/InspectFile.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+      // Construct request
+      InspectContentRequest request = InspectContentRequest.newBuilder()
+          .setParent(project.toString())
+          .setItem(item)
+          .setInspectConfig(config)
+          .build();
+
+      // Run request and parse response
+      InspectContentResponse response = dlp.inspectContent(request);
```
Comment (lesv) [inline dlp/src/main/java/dlp.snippets/InspectFile.java]: How long does this take?  Should I make it async?  Do it in it's own thread, or in a separate microservice?  How does this code guide my architecture choices?

File: dlp/src/test/java/dlp/snippets/InspectTests.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    System.setOut(null);
+    bout.reset();
+  }
+
+  @Test
+  public void testInspectString() {
+    InspectString.inspectString(PROJECT_ID,  "I'm Gary and my email is gary@example.com");
+
+    String output = bout.toString();
+    assertThat(output, CoreMatchers.containsString("Info type: PHONE_NUMBER"));
```
Comment (dzlier-gcp) [inline dlp/src/test/java/dlp/snippets/InspectTests.java]: Wait, is this verifying that the DLP API found a PHONE_NUMBER in that test string? Because I don't see one. If the output is only verifying that it was _looking_ for phone numbers, then verifying that it was also looking for email addresses doesn't verify that it found one...

Comment (kurtisvg) [reply dlp/src/test/java/dlp/snippets/InspectTests.java]: The snippet prints out any findings out to console, so this test verifies that at least one of the findings had the infotype `PHONE_NUMBER`. 

The test was incorrect to look for an `"EMAIL_ADDRESS"` - I had changed the test string but forgot to update the following asserts.

File: dlp/src/main/java/dlp.snippets/InspectFile.java (hunk 4 above)
Comment (kurtisvg) [reply dlp/src/main/java/dlp.snippets/InspectFile.java]: The snippets with longer running operations use PubSub actions for longer running jobs. Do you think that with a comparison between different snippets there is enough suggestions for architecture choices? If not, do you have suggestions on how to better convey it?

Comment (lesv) [reply dlp/src/main/java/dlp.snippets/InspectFile.java]: I'd be inclined to mention it in the preamble (class or method comments).

File: dlp/src/main/java/com/example/dlp/Inspect.java [hunk 6]
```diff
@@ -72,165 +72,6 @@
 
 public class Inspect {
```
Comment (tswast) [inline dlp/src/main/java/com/example/dlp/Inspect.java]: There's still a lot going on in this snippets class, including some command-line logic. Is this file intended to show the new sample format or just the other files?

File: dlp/src/main/java/dlp/snippets/InspectFile.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      // Construct the request to be sent by the client
+      InspectContentRequest request = InspectContentRequest.newBuilder()
+          .setParent(project.toString())
+          .setItem(item)
+          .setInspectConfig(config)
+          .build();
+
+      // Use the client to send the request and parse results
+      InspectContentResponse response = dlp.inspectContent(request);
+      System.out.println("Findings: " + response.getResult().getFindingsCount());
```
Comment (tswast) [inline dlp/src/main/java/dlp/snippets/InspectFile.java]: A little whitespace might between this and the above line might help show that line 79 is the one actually making the API request. In Python we had a pattern of doing

```
InspectContentResponse response = dlp.inspectContent(request);  // API request
```

to be 100% clear which lines are network calls. That might have been a pattern from when the client was less clear about what methods actually make requests, though.

File: dlp/src/main/java/com/example/dlp/Inspect.java (hunk 6 above)
Comment (kurtisvg) [reply dlp/src/main/java/com/example/dlp/Inspect.java]: I only ported over two of the samples into the new format - I'll move the rest over if/when the new format is finalized.

File: dlp/src/main/java/dlp/snippets/InspectFile.java (hunk 7 above)
Comment (kurtisvg) [reply dlp/src/main/java/dlp/snippets/InspectFile.java]: Good point. I've added the break and split the comments to make sure it is clear.

Comment (tswast) [reply dlp/src/main/java/dlp/snippets/InspectFile.java]: Thanks!

File: dlp/src/main/java/com/example/dlp/Inspect.java [hunk 8]
```diff
@@ -756,28 +597,7 @@ public static void main(String[] args) throws Exception {
...
-          filePath,
-          minLikelihood,
-          maxFindings,
-          infoTypesList,
-          customInfoTypesList,
-          includeQuote,
-          projectId);
-      // gcs file inspection
-    } else if (cmd.hasOption("gcs")) {
+    if (cmd.hasOption("gcs")) {
```
Comment (dzlier-gcp) [inline dlp/src/main/java/com/example/dlp/Inspect.java]: I don't know if this is in scope or just because the executable already exists in this project, but are we going to continue to surface executables with new samples? Last I heard it was "no," but just wanted to confirm.

Comment (kurtisvg) [reply dlp/src/main/java/com/example/dlp/Inspect.java]: No, the new format puts the nix on future CLIs

File: dlp/src/test/java/dlp/snippets/InspectTests.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (dzlier-gcp) [inline dlp/src/test/java/dlp/snippets/InspectTests.java]: Should we start using 2019 now?

**Comment by kurtisvg** [comment]
@dzlier-gcp Ready for another review.

**Comment by kurtisvg** [comment]
@dzlier-gcp ping

File: dlp/src/test/java/dlp/snippets/RedactTests.java [hunk 10]
```diff
@@ -16,6 +16,7 @@
 
 package dlp.snippets;
 
+// [START dlp_redact_image]
```
Comment (dzlier-gcp) [inline dlp/src/test/java/dlp/snippets/RedactTests.java]: Is this intentional? If it is there's no end tag.

Comment (kurtisvg) [reply dlp/src/test/java/dlp/snippets/RedactTests.java]: It was not intentional. Fixed.
